--- a/eiskaltdcpp-gtk/src/favoritehubs.cc
+++ b/eiskaltdcpp-gtk/src/favoritehubs.cc
@@ -45,7 +45,7 @@ FavoriteHubs::FavoriteHubs():
    // Fill the charset drop-down list in edit fav hub dialog.
    vector<string> &charsets = WulforUtil::getCharsets();
    for (vector<string>::const_iterator it = charsets.begin(); it != charsets.end(); ++it)
-       gtk_combo_box_text_append_text(GTK_COMBO_BOX_TEXT(getWidget("comboboxCharset")), it->c_str());
+       gtk_combo_box_append_text(GTK_COMBO_BOX(getWidget("comboboxCharset")), it->c_str());
 
    // Initialize favorite hub list treeview
    favoriteView.setView(GTK_TREE_VIEW(getWidget("favoriteView")), TRUE, "favoritehubs");
@@ -361,7 +361,7 @@ bool FavoriteHubs::showFavoriteHubDialog_gui(StringMap &params, FavoriteHubs *fh
 
        if (gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(fh->getWidget("checkbuttonEncoding"))))
        {
-           gchar *encoding = gtk_combo_box_text_get_active_text(GTK_COMBO_BOX_TEXT(fh->getWidget("comboboxCharset")));
+           gchar *encoding = gtk_combo_box_get_active_text(GTK_COMBO_BOX(fh->getWidget("comboboxCharset")));
            params["Encoding"] = string(encoding);
            g_free(encoding);
        }
