--- a/eiskaltdcpp-gtk/ui/search.ui
+++ b/eiskaltdcpp-gtk/ui/search.ui
@@ -190,11 +190,9 @@
                   </packing>
                 </child>
                 <child>
-                  <object class="GtkComboBox" id="comboboxentrySearch">
+                  <object class="GtkComboBoxEntry" id="comboboxentrySearch">
                     <property name="width_request">200</property>
                     <property name="visible">True</property>
-                    <property name="has_entry">True</property>
-                    <property name="entry_text_column">0</property>
                     <property name="can_focus">True</property>
                     <property name="model">liststore1</property>
                     <child>
@@ -203,6 +201,16 @@
                         <attribute name="text">0</attribute>
                       </attributes>
                     </child>
+                    <child internal-child="entry">
+                      <object class="GtkEntry" id="comboboxentry-entry1">
+                        <property name="visible">True</property>
+                        <property name="can_focus">True</property>
+                        <property name="primary_icon_activatable">False</property>
+                        <property name="secondary_icon_activatable">False</property>
+                        <property name="primary_icon_sensitive">True</property>
+                        <property name="secondary_icon_sensitive">True</property>
+                      </object>
+                    </child>
                   </object>
                   <packing>
                     <property name="expand">False</property>
@@ -325,8 +333,6 @@
                 <child>
                   <object class="GtkComboBox" id="comboboxFile">
                     <property name="visible">True</property>
-                    <property name="has_entry">True</property>
-                    <property name="entry_text_column">0</property>
                     <property name="can_focus">False</property>
                     <property name="model">liststore4</property>
                     <child>

