diff --git a/eiskaltdcpp-gtk/ui/search.ui b/eiskaltdcpp-gtk/ui/search.ui
index afc10f9..f5c6e4e 100644
--- a/eiskaltdcpp-gtk/ui/search.ui
+++ b/eiskaltdcpp-gtk/ui/search.ui
@@ -184,8 +184,6 @@
             <property name="visible">True</property>
             <property name="can_focus">False</property>
             <property name="model">liststore4</property>
-            <property name="has_entry">True</property>
-            <property name="entry_text_column">0</property>
             <child>
               <object class="GtkCellRendererText" id="cellrenderertext4"/>
               <attributes>
@@ -200,19 +198,27 @@
           </packing>
         </child>
         <child>
-          <object class="GtkComboBox" id="comboboxentrySearch">
+          <object class="GtkComboBoxEntry" id="comboboxentrySearch">
             <property name="width_request">200</property>
             <property name="visible">True</property>
             <property name="can_focus">True</property>
             <property name="model">liststore1</property>
-            <property name="has_entry">True</property>
-            <property name="entry_text_column">0</property>
             <child>
               <object class="GtkCellRendererText" id="cellrenderertext1"/>
               <attributes>
                 <attribute name="text">0</attribute>
               </attributes>
             </child>
+            <child internal-child="entry">
+              <object class="GtkEntry" id="comboboxentry-entry1">
+                <property name="visible">True</property>
+                <property name="can_focus">True</property>
+                <property name="primary_icon_activatable">False</property>
+                <property name="secondary_icon_activatable">False</property>
+                <property name="primary_icon_sensitive">True</property>
+                <property name="secondary_icon_sensitive">True</property>
+              </object>
+            </child>
           </object>
           <packing>
             <property name="expand">True</property>
