--- a/eiskaltdcpp-gtk/ui/settingsdialog.ui
+++ b/eiskaltdcpp-gtk/ui/settingsdialog.ui
@@ -1171,10 +1171,8 @@
                           </packing>
                         </child>
                         <child>
-                          <object class="GtkComboBoxText" id="comboboxCharset">
+                          <object class="GtkComboBoxEntry" id="comboboxCharset">
                             <property name="visible">True</property>
-                            <property name="has_entry">True</property>
-                            <property name="entry_text_column">0</property>
                             <property name="can_focus">False</property>
                             <property name="model">liststore1</property>
                             <child>
@@ -1328,10 +1326,8 @@
                           </packing>
                         </child>
                         <child>
-                          <object class="GtkComboBoxText" id="ipComboboxEntry">
+                          <object class="GtkComboBoxEntry" id="ipComboboxEntry">
                             <property name="visible">True</property>
-                            <property name="has_entry">True</property>
-                            <property name="entry_text_column">0</property>
                             <property name="can_focus">False</property>
                             <property name="model">liststore2</property>
                             <child>
